src/test/scala/com/cognite/sdk/scala/common/CdpApiExceptionSpec.scala

```diff
@@ -0,0 +1,95 @@
+package com.cognite.sdk.scala.common
+
+import org.scalatest.{FlatSpec, Matchers}
+import com.softwaremill.sttp._
+import io.circe._
+
+class CdpApiExceptionSpec extends FlatSpec with Matchers {
+  val url = uri"https://localhost"
+
+  it should "format messages without request ID" in {
+    val ex = CdpApiException(
+      url = url,
+      code = 404,
+      message = "Not Found",
+      missing = None,
+      duplicated = None,
+      missingFields = None,
+      requestId = None
+    )
+
+    ex.getMessage shouldBe s"Request to $url failed with status 404: Not Found."
+  }
+
+  it should "format messages with request ID" in {
+    val ex = CdpApiException(
+      url = url,
+      code = 400,
+      message = "Bad Request",
+      missing = None,
+      duplicated = None,
+      missingFields = None,
+      requestId = Some("1234")
+    )
+
+    ex.getMessage shouldBe s"Request with id 1234 to $url failed with status 400: Bad Request."
+  }
+
+  it should "format messages with request ID and duplicated" in {
+    val ex = CdpApiException(
+      url = url,
+      code = 400,
+      message = "Bad Request",
+      missing = None,
+      duplicated = Some(Seq(
+        JsonObject("id" -> Json.fromInt(1)),
+        JsonObject("externalId" -> Json.fromString("externalId-1")),
+        JsonObject("externalId" -> Json.fromString("externalId-2")),
+        JsonObject("id" -> Json.fromInt(2)),
+      )),
+      missingFields = None,
+      requestId = Some("1234")
+    )
+
+    ex.getMessage shouldBe
+      s"Request with id 1234 to $url failed with status 400: Bad Request. Duplicated externalIds: [externalId-1, externalId-2]. Duplicated ids: [1, 2]."
+  }
+
+  it should "format messages with request ID and duplicated and missing" in {
+    val ex = CdpApiException(
+      url = url,
+      code = 400,
+      message = "Bad Request",
+      missing = Some(Seq(
+        JsonObject("id" -> Json.fromInt(3))
+      )),
+      duplicated = Some(Seq(
+        JsonObject("id" -> Json.fromInt(1)),
+        JsonObject("externalId" -> Json.fromString("externalId-1")),
+        JsonObject("externalId" -> Json.fromString("externalId-2")),
```

Comment:
Switch the order on those to test the deterministic ordering.

Reply:
Good idea!